tests/admin_checks/tests.py

```diff
@@ -138,6 +143,86 @@ class SongAdmin(admin.ModelAdmin):
         ]
         self.assertEqual(errors, expected)
 
+    def test_wrong_prepopulated_fields(self):
+        class SongAdmin(admin.ModelAdmin):
+            prepopulated_fields = {'title': 'test'}
+
+        expected = [
+            checks.Error(
+                'The value of \'prepopulated_fields["title"]\' must be a list'
+                ' or tuple.',
+                obj=SongAdmin,
+                id='admin.E029',
+            )
+        ]
+        self.assertEqual(SongAdmin(Song, AdminSite()).check(), expected)
+
+    def test_wrong_readonly_fields(self):
+        class SongAdmin(admin.ModelAdmin):
+            readonly_fields = 'test'
+
+        expected = [
+            checks.Error(
+                "The value of 'readonly_fields' must be a list or tuple.",
+                obj=SongAdmin,
+                id='admin.E034',
+            )
+        ]
+        self.assertEqual(SongAdmin(Song, AdminSite()).check(), expected)
+
+    def test_wrong_list_editable(self):
+        class SongAdmin(admin.ModelAdmin):
+            list_editable = 'test'
+
+        expected = [
+            checks.Error(
+                "The value of 'list_editable' must be a list or tuple.",
+                obj=SongAdmin,
+                id='admin.E120',
+            )
+        ]
+        self.assertEqual(SongAdmin(Song, AdminSite()).check(), expected)
+
+    def test_list_editable_missing_field(self):
+        class SongAdmin(admin.ModelAdmin):
+            list_editable = ('test',)
+
+        expected = [
+            checks.Error(
+                "The value of 'list_editable[0]' refers to 'test', which is "
+                "not an attribute of 'admin_checks.Song'.",
+                obj=SongAdmin,
+                id='admin.E121',
+            )
+        ]
+        self.assertEqual(SongAdmin(Song, AdminSite()).check(), expected)
+
+    def test_both_list_editable_and_list_display_links(self):
+        class SongAdmin(admin.ModelAdmin):
+            list_editable = ('title',)
+            list_display = ('title',)
+            list_display_links = ('title',)
+
+        expected = [
+            checks.Error(
+                "The value of 'title' cannot be in both 'list_editable' and "
+                "'list_display_links'.",
+                obj=SongAdmin,
+                id='admin.E123',
+            )
+        ]
+        self.assertEqual(SongAdmin(Song, AdminSite()).check(), expected)
+
+    def test_missing_attributes(self):
```

Comment:
This test isn't needed. I removed the related code in 37520d284e3efb96eb24ba83ff77af56dc47adc3.